working-with-components.md

```diff
@@ -0,0 +1,50 @@
+
+ - **Atoms:** These are the basic building blocks of our pages. Atoms are our most simple components, such as a form label, a text input or a button.
+ - **Molecules:** These are groups of atoms bonded together and form larger reuseable components. These molecules take on their own emergent properties and can in turn be reused to build even more complex sections of a user interface. An example of a molecule component could be a label and input together.
+ - **Organisms:** These are groups of molecules joined together to form a relatively complex, distinct sections of our user interface. An example of an organism component may be a product tile, or a search header.
+
+At Holiday Extras we currently use Facebook's [React](https://facebook.github.io/react/index.html) for building new composeable user interface components, but these same atomic design principles apply equally well to other templating languages we use elsewhere - haml, handlebars, etc. 
+
+## Why should we use components?
+Building a library of reuseable components has several benefits:
+
+ - Once built, the shared components can be reused in a fraction of the time they took to build.
+ - Shared components provide a consistent look and feel to all of Holiday Extras' user interfaces.
+ - Shared components allow user interface improvements to be rapidly rolled out across our websites and all of our customers and partners to benefit in a fraction of the time they otherwise would.
+ - Simple reuseable components can be rapidly composed together to build more complex user interfaces.
+ - Reuseable components vastly reduce code duplication in the front end.
+
+In the short term there's an obvious implemntation cost to setting all this up, but taking a long term view and keeping the above in mind, the question is really *"Why wouldn't we use components?"*.
+
+## Where and when do we use components?
+In short, we'll use them everywhere. Our goal is to be integrating reuseable components across all of our user interfaces. From our landing pages and search engines, to our web and mobile apps. In order to do this we'd like everyone to look at componentising user interfaces as they work on them.
+
+As for when to use components, the answer to this question can be broken down into two sections:
+
+### Writing new user interface code
+This should be componentised from the beginning, and this is the expectation for new pull requests. The UX/UI Team are on hand to offer advice and assistance with this if necessary.
+
+### Modifying existing user interface code
+When modifying existing user interface code, existing user interface elements that can be replaced by shared components we already have should be replaced. If there are no existing user interface components we should look at building some when time allows.
+
```

Comment:
you could put a note in here about refactoring from the smallest components upwards. Always replace the end nodes of the tree (atoms) before attempting to refactor out molecules or organisms. Lesson learned from parking tile and wizard. Can like to Ryan Florence for more in depth info.